rocksdb/src/move.rs

```diff
@@ -0,0 +1,100 @@
+use {
+    bytes::Bytes,
+    eth_trie::DB,
+    move_binary_format::{errors::PartialVMError, CompiledModule},
+    move_core_types::{
+        account_address::AccountAddress,
+        language_storage::{ModuleId, StructTag},
+        metadata::Metadata,
+        resolver::{resource_size, ModuleResolver, ResourceResolver},
+        value::MoveTypeLayout,
+    },
+    move_table_extension::{TableHandle, TableResolver},
+    rocksdb::DB as RocksDb,
+};
+
+pub struct RocksEthTrieDb<'db> {
+    db: &'db RocksDb,
+}
+
+impl<'db> RocksEthTrieDb<'db> {
+    pub fn new(db: &'db RocksDb) -> Self {
+        Self { db }
+    }
+
+    pub fn db(&self) -> &'db RocksDb {
+        self.db
+    }
+}
+
+impl<'db> ModuleResolver for RocksEthTrieDb<'db> {
+    type Error = rocksdb::Error;
+
+    fn get_module_metadata(&self, module_id: &ModuleId) -> Vec<Metadata> {
+        if let Ok(Some(bytes)) = self.get_module(module_id) {
+            if let Ok(module) = CompiledModule::deserialize(&bytes) {
+                return module.metadata;
+            }
+        }
+
+        Vec::new()
+    }
+
+    fn get_module(&self, id: &ModuleId) -> Result<Option<Bytes>, Self::Error> {
+        let key = id.address().to_canonical_string();
+        let db = self.db();
+
+        db.get(key).map(|v| v.map(|v| Bytes::copy_from_slice(&v)))
+    }
+}
+
+impl<'db> ResourceResolver for RocksEthTrieDb<'db> {
+    type Error = rocksdb::Error;
+
+    fn get_resource_bytes_with_metadata_and_layout(
```

Comment:
Should we make this fn name less explicit like call it only `get_resource`?

Reply:
The name comes from the `ResourceResolver` trait, we cannot change it.